mapillary_tools/download_blurred.py

```diff
@@ -0,0 +1,413 @@
+import hashlib
+import os
+import requests
+import shutil
+import sys
+import threading
+import uploader
+import urllib
+from Queue import Queue
+import time
+import signal
+
+
+MAPILLARY_ENDPOINT = 'http://localhost:4000'
+MAPILLARY_API_IM_SEARCH_URL = "{}/v3/images?".format(MAPILLARY_ENDPOINT)
+
+
+class WorkerMonitor(threading.Thread):
+    """
+    This class should be used for logging state from currently running workers
+    against a Queue. Each worker is assumed to have `self.worker_stats`
+    property which is a number.
+    """
+
+    def __init__(self, threads, q):
+        threading.Thread.__init__(self)
+        self.threads = threads
+        self.q = q
+        self.shutdown_flag = threading.Event()
+
+    def run(self):
+        while not self.shutdown_flag.is_set():
+            stats = []
+            message = ''
+            total = 0
+            for index, worker in enumerate(self.threads):
+                stats = worker.worker_stats
+                total += stats
+                message += 'T{}: {}/{} '.format(index, stats, self.q.maxsize)
+
+            output = message + 'Total: {}/{}\r'.format(total, self.q.maxsize)
+            sys.stdout.write(output)
+            sys.stdout.flush()
+
+            if (self.q.empty()):
+                sys.stdout.write('\nDownload complete: {}\n'.format(self.q.maxsize))
+                self.shutdown_flag.set()
+
+
+def create_dirs(base_path):
+    """
+    Create directory if it doesn't exist.
+    :param base_path: where to create the directory
+    :type  base_path: string
+    """
+    try:
+        shutil.rmtree(base_path)
```

Comment:
that seems risky, what if clients specifies existing dir with other content, besides which he wants the images to be downloaded to, client could be in the home dir and pass `.` as output dir, that would be bad
if you want to clean a dir, then its best that you add a mapillary specific dir to the output path specified by the user, example os.path.join(output_folder, "mapillary_blurred_images")

Reply:
Thanks for spotting this, I'll remove this functionality and just create dirs when needed before event attempting the download.